### PR TITLE
Open dropdown before focus on submenu item.

### LINKED ISSUE #24806: Navigation screen: Clicking "Go to block" does not put focus into that block
Split out from https://github.com/WordPress/gutenberg/issues/24760.

Clicking the "Go to block" does reveal the nested selection in the Navigation block, but it does not put focus into that block.

### PR DESCRIPTION
**Pull request by tellthemachines**
## Description

Fixes #24806 .

In order to focus a block inside a dropdown, we need to open that dropdown first. This PR. makes the dropdowns open by first focusing on the parents of the target block. It's not super fast if there are multiple nesting levels, but I couldn't think of a better way to do it. 

## How has this been tested?

Tested in browser with keyboard.

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-editor/src/components/block-navigation/block.js [hunk 1]
```diff
@@ -46,20 +46,26 @@ export default function BlockNavigationBlock( {
...
 
-			return (
-				isBlockBeingDragged( clientId ) ||
-				isAncestorBeingDragged( clientId )
-			);
+			return {
+				isDragging:
+					isBlockBeingDragged( clientId ) ||
+					isAncestorBeingDragged( clientId ),
+				getParents: getBlockParents,
```
Comment (kevin940726) [inline packages/block-editor/src/components/block-navigation/block.js:61]: Couldn't we just extract `blockParents` here?

```diff
-				getParents: getBlockParents,
+				blockParents: getBlockParents( clientId ),
```

**Review by kevin940726 — DISMISSED** [review]
~Haven't tested it, just nitpicking some code smell 😝~

Tested it, it works fine! This still looks like a hacky way to do this though, but not that I have a better option 😝

**Review by draganescu — APPROVED** [review]
Tested and works great even with a larger nesting. cc @talldan do you have an idea of how to solve this differently?

**Comment by talldan** [comment]
@draganescu I think the submenus are causing a few issues along these lines. I've also noticed that you can use the left arrow key to navigate through each link individually, but when trying to use the right arrow key you end up stuck where the previous submenu is hidden:
arrowsubmenu

Potentially the block selection isn't the issue, but how the submenus are styled is. I notice currently they're `visibility: hidden`, but maybe they need to be visually hidden for keyboard navigation to work.

File: packages/block-editor/src/components/block-navigation/block.js (hunk 1 above)
Comment (tellthemachines) [reply packages/block-editor/src/components/block-navigation/block.js]: Done!

**Comment by talldan** [comment]
Hmm, I really don't think this was the right solution. How come my comment was completely ignored?

**Comment by tellthemachines** [comment]
Oh whoops, didn't see that!

> I've also noticed that you can use the right arrow key to navigate through each link individually, but when trying to use the left arrow key you end up stuck where the previous submenu is hidden:

This definitely needs to be fixed, but I'm not sure removing `visibility: hidden` is the solution, because if the hidden items become focusable, then we'll be able to go left from an item right into the previous item's submenu, which is probably not what the person tabbing or arrowing left is expecting at all.

**Comment by talldan** [comment]
> This definitely needs to be fixed, but I'm not sure removing visibility: hidden is the solution, because if the hidden items become focusable, then we'll be able to go left from an item right into the previous item's submenu, which is probably not what the person tabbing or arrowing left is expecting at all.

It's definitely a difficult thing to balance—unfortunately can't use space or something like that to open a submenu, so I wonder what the right way to navigate using the keyboard would be.

**Pull request merged**